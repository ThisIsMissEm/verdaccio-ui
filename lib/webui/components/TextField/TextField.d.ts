/**
 * @prettier
 */
import { TextFieldProps } from '@material-ui/core/TextField';
import React from 'react';
declare const TextField: React.FC<TextFieldProps>;
export default TextField;
