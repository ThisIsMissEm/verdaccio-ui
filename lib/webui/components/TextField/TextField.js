"use strict";
/**
 * @prettier
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const TextField_1 = __importDefault(require("@material-ui/core/TextField"));
const react_1 = __importDefault(require("react"));
const TextField = ({ InputProps, classes, ...other }) => (react_1.default.createElement(TextField_1.default, Object.assign({}, other, { InputProps: {
        ...InputProps,
        classes,
    } })));
exports.default = TextField;
