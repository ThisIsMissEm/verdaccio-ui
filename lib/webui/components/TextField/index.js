"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var TextField_1 = require("./TextField");
exports.default = TextField_1.default;
