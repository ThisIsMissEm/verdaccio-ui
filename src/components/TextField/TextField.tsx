import { default as TextFieldMaterialUI, TextFieldProps } from '@material-ui/core/TextField';
import React from 'react';

const TextField: React.FC<TextFieldProps> = props => <TextFieldMaterialUI {...props} />;

export default TextField;
