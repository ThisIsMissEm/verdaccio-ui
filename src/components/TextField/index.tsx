/**
 * @prettier
 */

import React from 'react';
import { TextFieldProps, default as TextFieldMui } from '@material-ui/core/TextField';

const TextField: React.FC<TextFieldProps> = ({ InputProps, classes, ...props }) => (
  <TextFieldMui
    {...props}
    InputProps={{
      ...InputProps,
      classes,
    }}
  />
);

export default TextField;
