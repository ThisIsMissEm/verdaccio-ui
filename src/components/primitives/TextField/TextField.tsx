import React from 'react'
import MuiTextField, { OutlinedTextFieldProps as MuiTextFieldProps } from '@material-ui/core/TextField'
import { usePyroField, PyroFormValues } from 'pyro-form'

interface Props<Values extends PyroFormValues> extends Omit<MuiTextFieldProps, 'variant'> {
  name: Extract<keyof Values, string>
}

const TextField = <Values extends PyroFormValues = any>({ name, ...muiProps }: Props<Values>) => {
  const { meta, core } = usePyroField<Values, Extract<keyof Values, string>>(name)
  return (
    <MuiTextField
      {...core}
      {...muiProps}
      error={meta.touched ? !!meta.error : undefined}
      helperText={meta.touched ? meta.error : undefined}
      variant="outlined"
    />
  )
}

export default TextField
