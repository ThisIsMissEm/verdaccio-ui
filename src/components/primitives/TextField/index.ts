/**
 * @prettier
 */

export { default } from './TextField';
