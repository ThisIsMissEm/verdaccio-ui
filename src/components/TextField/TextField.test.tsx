import React from 'react';
import { shallow } from 'enzyme';

import TextField from './TextField';

describe('<TextField /> component', () => {
  const props = {
    name: 'test',
    value: 'test',
  };
  test('should render the component in default state', () => {
    const wrapper = shallow(<TextField name={props.name} value={props.value} />);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
