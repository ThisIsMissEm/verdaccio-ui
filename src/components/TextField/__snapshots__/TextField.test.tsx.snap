// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextField /> component should render the component in default state 1`] = `"<div class=\\"MuiFormControl-root-1\\"><div class=\\"MuiInputBase-root-18 MuiInput-root-5 MuiInput-underline-9 MuiInputBase-formControl-19 MuiInput-formControl-6\\"><input type=\\"text\\" aria-invalid=\\"false\\" class=\\"MuiInputBase-input-28 MuiInput-input-13\\" name=\\"test\\" value=\\"test\\"/></div></div>"`;
